src/backend/distributed/metadata/distobject.c

```diff
@@ -160,6 +164,13 @@ MarkObjectDistributed(const ObjectAddress *distAddress)
 	{
 		ereport(ERROR, (errmsg("failed to insert object into citus.pg_dist_object")));
 	}
+
+	if (shouldSyncMetadata)
```

Comment:
To start with, we discussed doing a very similar thing for https://github.com/citusdata/citus/issues/3851 with @hanefi just last week.

If you look into the problem, they share a similar nature. We should sync certain object only to the nodes with metadata.

However, our idea was a little different. We considered to add a column to `pg_dist_object` such as `syncOnlyToMetadataNodes` because we thought that'd be a more natural way to integrate this concept, and handle `citus_add_node()` + `sync_metadata_to_node()` nicer.

Plus, we also discussed that it'd be simpler to expand this logic for views on distributed tables, as they also share the same problem.

How do you compare adding a column to `pg_dist_object` vs this one?

Reply:
I think you misunderstood the purpose of this flag. It's only to make sure that running `citus_internal_add_object_metadata` doesn't call this again. So it's purpose is similar to the `localOnly` flag you added in this PR: https://github.com/citusdata/citus/pull/5130

I will rename it to `localOnly` so the PRs naming scheme is aligned.

Reply:
I now understand what you mean. 

Still, wouldn't it be more generic to keep this information in the metadata tables? I have really hard time to follow the current approach of distributed sequences. We have bunch of special rules etc. And, for example, the complexity of this PR mostly because we do not have a nice way to deal with sequences.

Instead, I'd like to discuss a more generic/simple approach to this problem (which we can extend for functions/views etc):

- Add a new column to pg_dist_object: syncToMetadataNodes
-  Add a new API like `MarkObjectSyncToMetadataNodes` and call it only for objects that are required for the distributed table itself (such as sequences, views or functions)
- When `EnsureDependenciesExistOnAllNodes` is called, we treat `syncToMetadataNodes` objects slightly differently


I can understand that this PR can be merged without such a change, but I think we are even making the way sequences are treated more complex.

I think my suggestion would be to improve the handling of sequences in a separate PR (which we can probably work with @hanefi next week as you seem OOTO). Later, land this PR on top of that.

Let me know what you think